cmd/new.go

```diff
@@ -367,6 +371,29 @@ func errorIfNotEmptyDirectory(path string) error {
 	return nil
 }
 
+func validateProjectName(projectName string, opts display.Options) error {
```

Comment:
Note that the `pulumi new` command does accept a `--name` (or `-n`) flag and we [validate this early on](https://github.com/pulumi/pulumi/blob/1a698cbc9ed1ce8184bf31167e0a3b7fc82fadd1/cmd/new.go#L78-L81) and exit early if the specified project name isn't valid. But that validation isn't calling the new `DoesProjectExist`.

So even with the changes in this PR I could do:

```bash
pulumi new -n exists
```

And `pulumi` will happily continue and prompt for the project description and stack name, and will create the stack in the existing project assuming I choose a stack name that doesn't conflict with an existing stack name in the existing project.

I wonder if we should add _another_ check that calls this function to ensure the project doesn't exist earlier on. Perhaps after this block:

https://github.com/pulumi/pulumi/blob/1a698cbc9ed1ce8184bf31167e0a3b7fc82fadd1/cmd/new.go#L121-L127

Something like:

```go
// Ensure the project doesn't already exist.
if !generateOnly && name != "" {
	if err := validateProjectName(name, opts); err != nil {
		return err
	}
}
```

Reply:
However, when we get to the ideal fix for #2421 (prompting for the owner), we'll probably need to make some modifications to such a check as it'll need an owner to perform the "does exist" check. Perhaps it'd only do the check if it has an owner at this point (e.g. the user is logged in to the cloud backend and isn't a member of any orgs, and hence the user account would be the owner). Or if we decided to add an `--owner` flag and the flag was specified.